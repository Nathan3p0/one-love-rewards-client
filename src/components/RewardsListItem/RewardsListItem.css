.user__content-rewards--item {
    display: grid;
    grid-template-columns: 1fr 3fr;
    margin: 1rem;
  }

  .user__content-rewards--points {
    background-color: slategrey;
    padding: 1rem;
    text-align: center;
  }

  .user__content-rewards--reward {
    background-color: #bbb;
    text-align: center;
    padding: 1rem;
  }

  .user__content-rewards--earned {
    background-color: green;
  }