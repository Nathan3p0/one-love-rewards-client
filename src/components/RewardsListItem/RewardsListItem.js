import React from 'react'
import './RewardsListItem.css'

const RewardsListItem = (props) => {
    return (
        <li>
            <div className="user__content-rewards--item">
                <div className="user__content-rewards--points">
                    <p>{props.pointsRequired}pts</p>
                </div>
                <div className="user__content-rewards--reward">
                    <p>{props.discount}</p>
                </div>
            </div>
        </li>
    )
}

export default RewardsListItem