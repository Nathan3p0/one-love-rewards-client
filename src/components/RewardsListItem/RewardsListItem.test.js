import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import RewardsListItem from './RewardsListItem'

describe('Rewards List Item Component Rendering Tests', () => {

    it('renders without crashing', () => {
        const div = document.createElement('div')
        ReactDOM.render(<RewardsListItem />, div)
        ReactDOM.unmountComponentAtNode(div)
    })

    it('renders to UI as expected', () => {
        const tree = renderer.create(<RewardsListItem />).toJSON();
        expect(tree).toMatchSnapshot()
    })

    it('renders to UI as expected with PROPS', () => {
        const tree = renderer.create(<RewardsListItem button={true} currentPoints={250} pointsRequired={125} discount={'$25 Off'} onClickReward={() => { }} updated={0} />).toJSON();
        expect(tree).toMatchSnapshot()
    })

})