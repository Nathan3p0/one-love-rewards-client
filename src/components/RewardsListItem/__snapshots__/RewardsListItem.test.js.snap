// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rewards List Item Component Rendering Tests renders to UI as expected 1`] = `
<li>
  <div
    className="user__content-rewards--item"
  >
    <button
      className="user__content-rewards--points user__content-rewards--earned"
      onClick={[Function]}
    >
      <p>
        0
        pts
      </p>
    </button>
    <div
      className="user__content-rewards--reward"
    >
      <p>
        
      </p>
    </div>
  </div>
</li>
`;

exports[`Rewards List Item Component Rendering Tests renders to UI as expected with PROPS 1`] = `
<li>
  <div
    className="user__content-rewards--item"
  >
    <button
      className="user__content-rewards--points"
      onClick={[Function]}
    >
      <p>
        125
        pts
      </p>
    </button>
    <div
      className="user__content-rewards--reward"
    >
      <p>
        $25 Off
      </p>
    </div>
  </div>
</li>
`;
